### DISCUSSION TITLE
keep jump characters visible

### DISCUSSION THREAD

**Discussion by Magalaan** [junegunn/fzf#3281] [Ideas]
Jump allows selecting a line, pressing a key shown in the gutter. However, you need to activate jump first and make the characters shown, as key input normally goes to enter a search term. This is desired normal behavior. However, I use fzf a lot for creating menu's. I would like to keep the jump characters visible all time in menu's, 

I would like an option:  input-mode to allow keeping the jump keys visible,
--input-mode=find: default normal behavior like it is now
--input-mode=jump: keep jump-characters visible and accept selected line when in single-select and toggle line selection in case of multi-select 

the Jump key still toggles between accepting keys to jump or find

With only one key to press, this would allow very fast menu's. You can quickly move through menu's in a lotus 1-2-3 like manner, remembering key sequences.

**Comment by LangLangBart** [comment]
## jump/ jump-accept actions

- ref: added with [0.12.2](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0122)

```sh
man fzf
#            jump                         (EasyMotion-like 2-keystroke movement)
#            jump-accept                  (jump and accept)
```

## load event

- the `load` event does work nicely with both `jump` and `jump-accept`
  -  ref: added with [0.36.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0360)

```sh
fzf --bind 'load:jump'

fzf --bind 'load:jump-accept'
```

<img src="https://github.com/junegunn/fzf/assets/92653266/7c6aae06-ee66-495e-a4b7-3dd44cac7174" width="800">


```sh
man fzf
#        load
#               Triggered when the input stream is complete and the initial pro-
#               cessing of the list is complete.
```


---

## focus event

- I **wouldn't** use the `focus` event with `jump` because the `pointer` is always hidden behind the `jump` characters
  - ref: added with [0.37.0](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0370) 

```sh
fzf --bind 'focus:jump'
```

<img src="https://github.com/junegunn/fzf/assets/92653266/dd6c731f-df3c-49e5-a513-b692d619d214" width="800">

```sh 
man fzf
#        focus
#               Triggered  when the focus changes due to a vertical cursor move-
#               ment or a search result update.
```

---

PS:

- useful keys bindings for possible addition to the `FZF_DEFAULT_OPTS`
  - ref: https://github.com/junegunn/fzf/pull/569#issuecomment-220207020  (19/May/16) 

```sh
--bind ';:jump' 
--bind '::jump-accept' 
```
